IMP Inventory: update instructions and images

larm-odoo opened the pull request:
Task [id=3185334](https://www.odoo.com/web#id=3185334&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form)

Targets 14 only. Do not forward port to 15 or 16 (15 and 16 require additional information in the doc).  Updated images and changed second person language.

StraubCreative commented:
@hojo-odoo @Felicious @brse-odoo can one of you peer review this please?
When you're done tag @odoo/us-doc-review , thanks!

samueljlieber commented:
Attempted to fix build errors in https://github.com/odoo/documentation/commit/192c1c5e7e1ff5c80e7a18ae00b42163cf828002 and https://github.com/odoo/documentation/commit/e2a5630efcc95830ac8b08a4d06c3c87e57c7966. 

Error:
> sphinx-tabs 3.2.0 has requirement docutils~=0.16.0, but you'll have docutils 0.14 which is incompatible. 

My docutils version is `0.16` - maybe the PR author needs to make this commit with the correct docutils version 🤔 I will work with @larm-odoo to get this resolved!

StraubCreative commented:
> Attempted to fix build errors in [192c1c5](https://github.com/odoo/documentation/commit/192c1c5e7e1ff5c80e7a18ae00b42163cf828002) and [e2a5630](https://github.com/odoo/documentation/commit/e2a5630efcc95830ac8b08a4d06c3c87e57c7966).
> 
> Error:
> 
> > sphinx-tabs 3.2.0 has requirement docutils~=0.16.0, but you'll have docutils 0.14 which is incompatible.
> 
> My docutils version is `0.16` - maybe the PR author needs to make this commit with the correct docutils version 🤔 I will work with @larm-odoo to get this resolved!

ANV mentioned this isn't on us in #documentation channel. 
Can proceed as normal.

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
-*FEFO* strategy selected, a transfer is requested for the five boxes of 20002 and one from 20001.
-The transfer for all the boxes of the lot 20002 is because they have the closest expiration date.
-The transfer also requests one box from 20001 because it’s the lot that expires the sooner after 20002.
+Basically, for every sales order of a product with the FEFO strategy, a transfer is requested for
+the product that has the earliest expiration date from the order date.
```
```suggestion
Using the :abbr:`FEFO (First Expiry, First Out)` removal strategy, every sales order that includes products with this removal strategy assigned ensures that transfers are requested for products with the expiration date soonest to the order date.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
-.. image:: removal/reserved-lots-FIFO.png
-   :align: center
-   :alt: two lots being reserved for sell with the FIFO strategy.
+Once the sales order is confirmed, the delivery order will be created and linked to the picking, and
+the oldest lot numbers will be reserved thanks to the FIFO strategy.
```
I think to beef up this section we could walk through the whole SO process. After we create and confirm it, go to the picking, then to the warehouse delivery form, and deliver/invoice for the products.

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
 
-With a *LIFO* strategy, that’s quite the opposite. In fact, the products that are brought the
-last move out the first. It is mostly used in case of products without a shelf life.
+With a :guilabel:`LIFO` strategy, the opposite is true. The products that are received last move out
+first. :guilabel:`LIFO` is mostly used for products without a shelf life.
```
```suggestion
As explained earlier, the :abbr:`LIFO (Last In, First Out)` removal strategy works in the **opposite** manner from the :abbr:`FIFO (First In, First Out)` strategy. With this method, the products that are received **last** are intended to move out first. This method is mostly used for products without a shelf life, and no time-sensitive factors, such as expiration dates.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
-Even if our white shirts are clothes, we can say that they are timeless. So, let’s use them to
-test our *LIFO* strategy. Once again, open the product category via :menuselection:`Inventory
---> Configuration --> Product Categories` and change the removal strategy to *LIFO*.
+In this example, the white shirts will be used again to test the :guilabel:`LIFO` strategy. First,
+open the product category via :menuselection:`Inventory --> Configuration --> Product Categories`
+and change the :guilabel:`Force Removal Strategy` to :guilabel:`LIFO`.
```
```suggestion
To test this example using the :abbr:`LIFO (Last In, First Out)` strategy, first navigate to :menuselection:`Inventory app --> Configuration --> Product Categories`, and select a product category to edit. This reveals a product category form.

Once on the product category form, under the :guilabel:`Logistics` section, change the :guilabel:`Force Removal Strategy` to :guilabel:`Last In First Out (LIFO)`.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
-.. image:: removal/reserved-lots-LIFO.png
-   :align: center
-   :alt: two lots being reserved for sell with the LIFO strategy.
+Then, :guilabel:`Create` a sales order for four white shirts and :guilabel:`Confirm` it. Check that
```
I think we can replicate the sales order to delivery process here, as well

larm-odoo replied:
Same as above. I did do one walkthrough of a SO in the first comment you made for adding it, so I think that might also cover all the subsequent SO examples.

brse-odoo replied:
I agree with you here and think explaining it earlier suffices 👍

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
+Operations` in the  :guilabel:`Sales Order`.
 
 .. important::
-   Don’t forget that the *LIFO* strategy is banned in many countries!
+   Don't forget that the :abbr:`LIFO (Last In, First Out)` strategy is banned in many countries!
```
I think we can delete this note, since it was mentioned a few sections up. I think saying it once with a `:warning:` tag is sufficient

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
 
-With the *FEFO* strategy, the way products are picked is not based on the reception date. In this
-particular case, they are dispatched according to their expiration date.
+With the :guilabel:`FEFO` removal strategy, the way products are picked is not based on the
+reception date. In this case, they are picked according to their expiration date.
```
```suggestion
As explained earlier, the :abbr:`FEFO (First Expiry, First Out)` removal strategy differs from the :abbr:`FIFO (First In, First Out)` and :abbr:`LIFO (Last In, First Out)` strategies, because it targets products for removal based on **expiration dates** instead of their warehoues receipt dates.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
 
 .. note::
-   To have more information about Expiration date, please have a look at
+   For more information about expiration dates, please have a look at
```
```suggestion
   For more information about expiration dates, reference
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
-   Other dates are for informational and reporting purposes only. If not removed from the stock,
-   lots that are past the expiration dates may still be picked for delivery orders!
+   If products are not removed from stock when they should be, lots that are past the expiration
+   date may still be picked for delivery orders!
```
```suggestion
   date may still be picked for delivery orders.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
 Lots are picked based on their removal date, from earliest to latest. Lots without a removal date
 defined are picked after lots with removal dates.
 
 .. note::
```
```suggestion
.. important::
```
Because we are dealing with perishable products :-)

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,245 @@
-   On the transfer form, you can find the product’s lot/serial number to pick for delivery.
+   To pick for delivery, the product's lot/serial number can be found on the transfer form.
 
 How does it work?
```
i like it!

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,245 @@
 How does it work?
 =================
 
 First In, First Out (FIFO)
```
I also agree- because these are how they are listed in the app as well, so for consistency, I thought it should also be like this.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,245 @@
 .. note::
-   This strategy is banned in many countries and can lead to only have old or obsolete products
-   in your stock.
+   The LIFO strategy is banned in many countries and can lead to only having old or obsolete products
+   in stock.
```
I looked around and didn't see anything. I'm going to ask the PO if there is anything related to this we can link to, if not, I'll cut it off where you suggested.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,245 @@
-
-So, for every order of a product with the *FIFO* strategy selected, Odoo requests a transfer for the
-good that has been in your stock for the longest period.
+Lot 00001 entered the stock on May 23, lot 00002 on May 25, and lot 00003 on June 1. A customer
+orders six boxes on June 11. With the FIFO removal strategy selected, a transfer is requested for
+the five boxes from lot 00001 and one of the boxes in lot 00002, since lot 00001 entered the stock
+first. The box from lot 00002 is taken next because it has the oldest receipt date after lot 00001.
+So, for every order of a product with the FIFO strategy selected, Odoo requests a transfer for the
+products that have been in stock for the longest time.
```
I added more details- I ma not sure if I went too far and added too much information, but the image is definitely good since it shows the lot numbers being picked.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
 
-First Expire, First Out (FEFO)
-------------------------------
+For example, imagine there are three lots of six-egg boxes (in this specific case, don't forget to
+use units of measure). Those three lots have the following lot numbers: 20001, 20002, and 20003,
+each with five boxes in it.
```
@brse-odoo I am having an issue getting this to work- I set my category to FEFO, and I just can't get my database to give me the dates. Could you help with this image?

larm-odoo replied:
I guess I just needed a few weeks to reset my brain- I was able to do this, it worked! I added the image.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
 
-The **First Expire, First Out** strategy is a bit different from the two others. Here, it is the
-expiration date that is important and not the date the product entered the stock.
+Lot 20001 entered the stock on July 1st and expires on July 15th, lot 20002 entered on July 2nd and
+expires on July 14th, and lot 20003 entered on July 3rd and expires on July 21st. A customer orders
+six boxes on July 5th. With the FEFO strategy selected, a transfer is requested for the five boxes
+from lot 20002 and one from lot 20001. All the boxes in lot 20002 are transferred because they have
+the earliest expiration date. The transfer also requests one box from lot 20001 because it has the
+next closest expiration date after lot 20002.
```
Same with this- I couldn't get the eggs to enter correctly, so I can't show that or the image above.

larm-odoo replied:
Same with this- seems like I just needed a few week reset then everything worked =D

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,237 @@
 
-On the delivery order linked to the picking, you can see that the oldest lot numbers have been
-reserved thanks to the *FIFO* strategy.
+As seen above, lot 000001 entered the stock first. :guilabel:`Create` and :guilabel:`Confirm` a
```
My only issue with this is, we don't also go through all the directions of receiving in products with the SN's- and you won't be able to make a sales order without the items sorted correctly. So while we could add all of this, we'd also have to add that- and I am worried this doc is going to get far too long. Now I am wondering if I should have made separate docs for all removal strategies separately, and gone through the whole 'receive the product and sell the product' for each one. Maybe that could be another revision- just an idea (good idea for the net versions.. incremental changes and all =).

larm-odoo replied:
Update: I added the sales order instructions, and since it's just to get them to understand how to do that part, I think it's fine.

larm-odoo commented:
Hi @brse-odoo - I had a few issues while doing these edits. I had one issue with the eggs that I couldn't make work and was hoping you could help me with that. Also, I added a few comments to the SO issues- can you let me know what you think? I am not sure if we should include ALL the instructions for those sections, or if we should reference other docs instead of making this a huge beast/monster of a doc.

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
-with the closest date first. Depending on the removal strategy you chose, Odoo generates a transfer
-with the products fitting your settings the best.
+Next, several orders for the same product are made, but in this example, the goods weren't received
+on the same day and they don't have the same expiration date. In that situation, logically, sending
```
```suggestion
on the same day, and they don't have the same expiration date. In that situation, sending
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
-with the products fitting your settings the best.
+Next, several orders for the same product are made, but in this example, the goods weren't received
+on the same day and they don't have the same expiration date. In that situation, logically, sending
+those with the closest expiration date is preferred. Depending on the chosen removal strategy, Odoo
```
```suggestion
the products with the closest expiration date is preferred. Depending on the chosen removal strategy, Odoo
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+Next, several orders for the same product are made, but in this example, the goods weren't received
+on the same day and they don't have the same expiration date. In that situation, logically, sending
+those with the closest expiration date is preferred. Depending on the chosen removal strategy, Odoo
+generates a transfer with the products that fit the settings best.
```
```suggestion
generates a transfer with the products that best fit the settings.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+boxes from lot 00001 and one of the boxes in lot 00002, since lot 00001 entered the stock first. The
+box from lot 00002 is taken next because it has the oldest receipt date after lot 00001.
+
+So, for every order of a product with the FIFO strategy selected, Odoo requests a transfer for the
```
```suggestion
So, for every order of a product with the :abbr:`FIFO (First In, First Out)` strategy selected, Odoo requests a transfer for the
```
I'd keep the abbreviation tags consistent when referencing the removal strategies

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+in the far right for the product in question. The :guilabel:`Detailed Operations` window appears,
+and displays the lot numbers selected for that specific product for the delivery order.
+
+Below displays the lot numbers for the nails example:
```
```suggestion
```
Think we can delete this line

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+:menuselection:`Inventory app --> Products  --> Lots/Serial Numbers`. This reveals a page with
+drop-down menus of all products assigned lots or serial numbers, filtered by *product* by default.
+To change the category these products are filtered by, click :guilabel:`Product` (in the search bar,
+in the top right of the page) to remove the default filter, and select a new filter (if desired).
```
```suggestion
in the top right of the page) to remove the default filter, and select a new filter if desired.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+
+To differentiate some units of products from others, the units need to be tracked, either by
+:guilabel:`Lot` or by :guilabel:`Serial Number`. To do so, go to :menuselection:`Inventory -->
+Configuration --> Settings`. Then, activate the :guilabel:`Storage Location`, :guilabel:`Multi-Step
```
```suggestion
Configuration --> Settings`. Then, activate the :guilabel:`Storage Locations`, :guilabel:`Multi-Step
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+Lot 000001 contains five shirts, lot 000002 contains three shirts, and lot 000003 contains two
+shirts.
+
+Go to the *Sales* app and click :guilabel:`Create` to create an :abbr:`SO (Sales Order)`. Next,
```
```suggestion
Go to the :menuselection:`Sales` app and click :guilabel:`Create` to create an :abbr:`SO (Sales Order)`. Next,
```
I *think* we use menuselection tags when we are telling someone to "go somewhere" on the screen

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
-On the delivery order linked to the picking, you can see that the oldest lot numbers have been
-reserved thanks to the *FIFO* strategy.
+Once the sales order is confirmed, the delivery order will be created and linked to the picking, and
+the oldest lot numbers will be reserved thanks to the FIFO strategy. All five shirts from lot 000001
```
```suggestion
the oldest lot numbers will be reserved thanks to the :abbr:`FIFO (First In, First Out)` strategy. All five shirts from lot 000001
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
-numbers to be used in *FEFO*. These dates can be set by going to :menuselection:`Inventory -->
-Master Data --> Lots/Serial Numbers`.
+Once the expiration dates feature is activated, it's possible to define different expiration dates
+for individual serialized products, **or** lot numbers containing many products.
```
```suggestion
for individual serialized products, as well as lot numbers containing many products.
```

brse-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
-   :alt: view of the FEFO strategy being set up as forced removal strategy.
-
-For this particular case, let’s use hand cream. As usual, we have three lots of them.
+.. important::
```
```suggestion
.. warning::
```
 I would argue this gets a warning tag, since sending expired products to customers could be pretty serious

larm-odoo replied:
very good idea- I agree!

brse-odoo reviewed (commented):
Hi @larm-odoo, just finished my review! Great job, let me know if you have any questions about my comments. Thanks!

meng-odoo reviewed (changes requested):
Hi @larm-odoo, great job on this doc! It's very thorough and contains a lot of information that is explained well. I had some suggestions for organization and trimming down the "Using removal strategies" section. Please let me know if you have any questions, and then please tag me for a second review :) thanks!

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
 
 .. note::
-   On the transfer form, you can find the product’s lot/serial number to pick for delivery.
+   To pick for delivery, the product's lot/serial number can be found on the transfer form.
```
Picking is one of the main functions/aspects of the inventory app, so I think that people reading this should understand what that means.  The transfer form is created once a delivery order is created, and then a picking/transfer is created to get the products form the warehouse to the shipping area - again, I think something that will be understood. I can add a link to another doc to make this more understood

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+
+To view the serial numbers being selected for a sales order, go to the :guilabel:`Sales app` and
+click the sales order in question. In the sales order, click the :guilabel:`Delivery` smart button
+in the top right. In the :guilabel:`Operations` tab, click the :guilabel:`Detailed Operations` icon
```
I tried to find an emoji or icon  like this but can't, and I checked with Sam, too. If you know of one, I'd be happy to use it, but I can't seem to find anything appropriate. I don't want to add too many images since I add a lot already, and I think the description is enough.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
-For this example, we created three lots of white shirts. Those are from the All/Clothes category,
-where we put *FIFO* as the removal strategy. In our stock location (WH/Stock), we now find the
-three lots available.
+In this example, there are three lots of white shirts. The shirts are from the All/Clothes category,
```
I like keeping the examples only because it fleshes things out a lot for the reader. I know it's a long doc, but I think it does add value.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
 
-On the delivery order linked to the picking, you can see that the oldest lot numbers have been
-reserved thanks to the *FIFO* strategy.
+Go to the :menuselection:`Sales app` and click :guilabel:`Create` to create an :abbr:`SO (Sales
+Order)`. Next, select a :guilabel:`Customer` from the drop-down menu. Then click :guilabel:`Add a
+product` in the :guilabel:`Order Lines` tab. Select the :guilabel:`White Shirt` from the drop-down
+menu, or type in the name of the product in the field. Enter `6.00` in the :guilabel:`Quantity`
+field, then click :guilabel:`Save`, then click :guilabel:`Confirm`.
```
I feel keeping it in gives them a more fleshed out example for this, and helps. Especially if they are unfamiliar with the sales app, having this paragraph could help or reinforce the steps, especially since these docs are very useful to people new to Odoo (I think) and it could help connect some dots they might not have connected. I see your point, but I do like having these fuller examples with each 'this is how it works'.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+So, for every order of a product with the :abbr:`FIFO (First In, First Out)` strategy selected, Odoo
+requests a transfer for the products that have been in stock for the longest time.
+
+To view the serial numbers being selected for a sales order, go to the :guilabel:`Sales app` and
```
I thought about this, but I also remember thinking about people who would go to a doc and only read that section they needed. So because of that, I like having the small Odoo example in the 'how it works' section. So it gives an example of how it would work, and a little info on how you would check things (like SN's). Then the in-depth is below.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,352 @@
+In Odoo, this is done either in the *Inventory* or *Barcode* app. In either app, in the default
+Kanban view, click or tap :guilabel:`Receipts`, then, click or tap on the individual receipt to
+validate it. Next, in the *Inventory* app, click :guilabel:`Edit`, then enter the received quantity
+in the :guilabel:`Done` column. Finally, click :guilabel:`Validate` to receive the products. If
```
It's odd, but you don't have to- I did the flow again, and you just need to enter the SN and click Validate, and it 'takes' it. I found my item in the inventory SN list report, so it takes it without clicking Save

larm-odoo commented:
Thanks  @meng-odoo for the review! I took most of the suggestions, except for the removing examples part, and I explained why in my comments. Otherwise, I implemented everything else. Ready for another look!

meng-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,363 @@
+   date may still be picked for delivery orders!
 
-For this particular case, let’s use hand cream. As usual, we have three lots of them.
+To test this example using the :abbr:`LIFO (Last In, First Out)` strategy, first navigate to
```
Same here---the example should be set up again in this section. Also should be "FEFO" instead of "LIFO" here :)

meng-odoo reviewed (changes requested):
Hi @larm-odoo, thanks for making those changes and responding to my comments! I made some different organizational suggestions to the doc in light of your responses. Other than that, I just had small wording suggestions. Please let me know if you have any questions!

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,363 @@
+.. image:: removal/lot-serial.png
    :align: center
-   :alt: features to enable in order to properly use removal strategies.
+   :alt: Click on Products, then Lots/Serial Numbers to display all the products with lots or serial
+         numbers.
 
-.. note::
-   To work with the *FEFO* strategy, activate the *Expiration Dates* feature.
+To change the category these products are filtered by, click :guilabel:`Product` (in the search bar,
+in the top right of the page) to remove the default filter, and select a new filter (if desired).
 
-Next, you need to define your removal strategy, on *Product Categories* via
-:menuselection:`Inventory --> Configuration --> Product Categories`.
+To view the expiration dates of all products, click the :guilabel:`⋮ (additional options icon)`
+(three-dots) on the far right of the page, and click the checkbox next to :guilabel:`Expiration
+Date`. Then, locate the :guilabel:`Expiration Date` column in any of the drop-downs on the page.
 
-.. image:: removal/first-in-first-out.png
+.. image:: removal/expiration.png
    :align: center
-   :alt: force removal strategy set up as first in first out.
+   :alt: Click the additional options icon, then click the checkbox next to Expiration Date to make
+         the column visible.
```
I rearranged it- I think it makes sense now.

larm-odoo commented:
Hi @meng-odoo - after taking some more suggestions from your review, I ended up cleaning up the 'how they work' section and beefing up the 'using it' sections- so you'll see more instructions in the 'using it' section, and less screenshots in the 'how it works' section. I think it's more of what you had in mind to begin with, and I think is more clear. Thank you very much- I think we are in a much better place with the doc now!

meng-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,377 @@
+contains four cinder blocks.
+
+To see how the :abbr:`LIFO (Last In, First Out)` strategy works, first navigate to
+:menuselection:`Inventory app --> Configuration --> Product Categories`, and select the All/Building
+Materials category to edit. This reveals a product category form.
```
```suggestion
:menuselection:`Inventory app --> Configuration --> Product Categories`, and select a product category (for this example, the All/Building
Materials category) to edit. This reveals a product category form.
```

meng-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,377 @@
-   Other dates are for informational and reporting purposes only. If not removed from the stock,
-   lots that are past the expiration dates may still be picked for delivery orders!
+To see how the :abbr:`FEFO (First Expiry, First Out)` strategy works, first navigate to
+:menuselection:`Inventory app --> Configuration --> Product Categories`, and select the All/Health &
+Beauty category to edit. This reveals a product category form.
```
```suggestion
:menuselection:`Inventory app --> Configuration --> Product Categories`, and select a product category (for this example, the All/Health &
Beauty category) to edit. This reveals a product category form.
```

meng-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,377 @@
+of hand cream, expiring on November 30, and lot 0000003 contains ten tubes of hand cream, expiring
+on October 31.
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Lot/Serial No
+     - Product
+     - Expiration Date
+     - Amount In Stock
+   * - 0000001
+     - Hand Cream
+     - 09/30
+     - 20
+   * - 0000002
+     - Hand Cream
+     - 11/30
+     - 10
+   * - 0000003
+     - Hand Cream
+     - 10/31
+     - 10
```
This table is great, but it feels out of place here since it's the only table in the doc, and all the other examples use images instead of tables. I would replace this with an image for consistency.

meng-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,377 @@
-+-----------------------+---------------+-----------------------+
-| 0000003               | Hand Cream    | 10/31/2019            |
-+-----------------------+---------------+-----------------------+
+Next, go to the :menuselection:`Sales app` and click :guilabel:`Create` to create an :abbr:`SO
+(Sales Order)`. Next, select a :guilabel:`Customer` from the drop-down menu. Then click
+:guilabel:`Add a product` in the :guilabel:`Order Lines` tab. Select the :guilabel:`Hand Cream` from
+the drop-down menu, or type in the name of the product in the field. Enter `25.00` in the
+:guilabel:`Quantity` field, then click :guilabel:`Save`, then click :guilabel:`Confirm`.
```
```suggestion
Next, go to the :menuselection:`Sales app` and click :guilabel:`Create` to create a sales
order. Next, select a :guilabel:`Customer` from the drop-down menu. Then click
:guilabel:`Add a product` in the :guilabel:`Order Lines` tab. Select a product (for this example, the :guilabel:`Hand Cream`) from
the drop-down menu, or type in the name of the product in the field. Enter a quantity (for example, `25.00`) in the
:guilabel:`Quantity` field, then click :guilabel:`Save`, then click :guilabel:`Confirm`.
```

meng-odoo reviewed (changes requested):
Hi @larm-odoo, this doc looks great! Very well-organized and well-written. I just had a few more tiny notes/suggestions/corrections, but please go ahead and tag Sam for the next review stage after you address those comments. Great job on this!

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,377 @@
+products for removal based on date of entry into the warehouse, the :guilabel:`First Expired, First
+Out (FEFO)` method targets products for removal based on their assigned expiration dates.
+
+Using the :abbr:`FEFO (First Expiry, First Out)` removal strategy, every sales order that includes
```
Thanks- I was torn because I like to be consistent with the app, and match the words so people don't get confused. I prefer "expired" (sounds better!)

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+
+To view the serial numbers being selected for a sales order, go to the :guilabel:`Sales app` and
+select the sales order in question. In the sales order, click the :guilabel:`Delivery` smart button
+in the top right. In the :guilabel:`Operations` tab, click the :guilabel:`⋮≣ (Detailed Operations)`
```
I think this is ok to use two icons to form the UI icon, although I think the one with 4 dots and 4 lines is a closer match
```suggestion
in the top right. In the :guilabel:`Operations` tab, click the :guilabel:`⦙≣ (Detailed Operations)`
```

larm-odoo replied:
Thank you! I think KSC found the icon for me since I couldn't find this one on my own (I still don't know how people find icons I can't seem to find).

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
-For this example, we created three lots of white shirts. Those are from the All/Clothes category,
-where we put *FIFO* as the removal strategy. In our stock location (WH/Stock), we now find the
-three lots available.
+In this example, there are three lots of white shirts. The shirts are from the All/Clothes category,
+where :guilabel:`FIFO` is set as the removal strategy. In the :guilabel:`Inventory Valuation
+Report`, the three different receipts are listed with the amounts.
```
The first half of this example is not part of an active instruction, so using italics rather than guilabel

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+**last** are moved out first. This method is mostly used for products without a shelf life, and no
+time-sensitive factors, such as expiration dates.
+
+In this example, there are three lots of cinder blocks. The blocks are from the All/Building
+Materials category, where :guilabel:`FIFO` is set as the removal strategy. In the
+:guilabel:`Inventory Valuation Report`, the three different receipts are listed with the amounts.
```
The first half of this example is not part of an active instruction, so using italics rather than guilabel
```suggestion
In this example, there are three lots of cinder blocks. The blocks are from the *All/Building
Materials* category, where *FIFO* is set as the removal strategy. In the :guilabel:`Inventory
Valuation Report`, the three different receipts are listed with the amounts.
```

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
 .. note::
-   To have more information about Expiration date, please have a look at
-   :doc:`the related doc <../../management/lots_serial_numbers/expiration_dates>`.
+   For more information about expiration dates, reference the
+   :doc:`Expiration dates <../../management/lots_serial_numbers/expiration_dates>` document.
```
Can use global relative path here
```suggestion
   For more information about expiration dates, reference the :doc:`Expiration dates
   </applications/inventory_and_mrp/inventory/management/lots_serial_numbers/expiration_dates>`
   document.
```

larm-odoo replied:
I didn't realize I could break up that line as well - I thought it had to be continuous.

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
-.. image:: removal/removal-date.png
-   :align: center
-   :alt: view of the removal date for 0000001.
+In this example, there are three lots of hand cream. The creams are from the All/Health & Beauty
+category, where :guilabel:`FEFO` is set as the removal strategy. In the :guilabel:`Inventory
```
The first half of this example is not part of an active instruction, so using italics rather than guilabel
```suggestion
In this example, there are three lots of hand cream. The creams are from the *All/Health & Beauty*
category, where *FEFO* is set as the removal strategy. In the :guilabel:`Inventory Valuation
```

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+
+To see how the :abbr:`LIFO (Last In, First Out)` strategy works, first navigate to
+:menuselection:`Inventory app --> Configuration --> Product Categories`, and select a product
+category (for this example, the All/Building Materials category) to edit. This reveals a product
```
Add :guilabel: to the element with active instruction
```suggestion
category (for this example, the :guilabel:`All/Building Materials` category) to edit. This reveals a
```

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
-For this particular case, let’s use hand cream. As usual, we have three lots of them.
+To see how the :abbr:`FEFO (First Expired, First Out)` strategy works, first navigate to
+:menuselection:`Inventory app --> Configuration --> Product Categories`, and select a product
+category (in this example, the All/Health & Beauty category) to edit. This reveals a product
```
Add :guilabel: to the element with active instruction
```suggestion
category (in this example, the :guilabel:`All/Health & Beauty` category) to edit. This reveals a
product category form.
```

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+leaving the warehouse might go through each of these locations at some point, removal strategies can
+have an effect on which products are taken, from where, and when.
 
 .. image:: removal/empty-dock.png
```
Image needs to be resized

larm-odoo replied:
I have NO idea how these got messed up! I retook them and I *know* I made them 933.... yikes. I'm sorry- I don't understand what happened (unless I saved them on my local instead of the github image folder!) Apologies- I know better than this =)

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+:guilabel:`Validate`. After products are received in Odoo, the products can be moved to their
+respective storage locations.
 
 .. image:: removal/entering-stocks.png
```
Image needs to be resized

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+the products with the closest expiration date is preferred. Depending on the chosen removal
+strategy, Odoo generates a transfer with the products that best fit the settings.
 
 .. image:: removal/packing-products.png
```
Image needs to be resized

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+where :guilabel:`FIFO` is set as the removal strategy. In the :guilabel:`Inventory Valuation
+Report`, the three different receipts are listed with the amounts.
 
 .. image:: removal/inventory-valuation.png
```
Image needs to be resized

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+All five shirts from lot 000001 and one shirt from lot 000002 will be selected to be sent to the
+customer.
 
 .. image:: removal/reserved-lots-FIFO.png
```
Image needs to be resized

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+Once on the product category form, under the :guilabel:`Logistics` section, change the
+:guilabel:`Force Removal Strategy` to :guilabel:`Last In First Out (LIFO)`.
 
 .. image:: removal/last-in-first-out.png
```
Image needs to be resized

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+date in the :guilabel:`Dates` tab. Finally, click :guilabel:`Save`.
 
-.. image:: removal/first-expiry-first-out.png
+.. image:: removal/removal-date.png
```
Image needs to be resized

samueljlieber reviewed (changes requested):
Hi @larm-odoo 👋 this is a very thorough doc! Nice work! I have a handful of changes and suggestions from my technical review, please see below.

I will also push up a second commit with only the resized images that I commented on so you can include them with the rest of changes. Remember to pull using `git pull --rebase` 🙂

Tag me for another look once this pr is ready! Thank you!

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,361 @@
+that, products are stored in their respective locations.
+
+In Odoo, receiving products is done either in the *Inventory* or *Barcode* app. In either app, in
+the default Kanban view, click :guilabel:`Receipts`, then, click on the individual receipt. Next, in
```
Oh, lol- I thought "Kanban" was a proper name - all I know is it's Japanese, hehe.

larm-odoo commented:
Hi Sam- all set! I think I did the rebase properly- all changes and images are updated, looked good in the 'make html' preview.

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
-   On the transfer form, you can find the product’s lot/serial number to pick for delivery.
+   To pick for delivery, the product's lot/serial number can be found on the transfer form. To learn
+   more about picking and shipping, refer to either
+   :doc:`../../management/shipments_deliveries/receipts_delivery_two_steps`
+   or :doc:`../../management/shipments_deliveries/delivery_three_steps`.
```
Let's move "or" since it is an early line break :)
```suggestion
   :doc:`../../management/shipments_deliveries/receipts_delivery_two_steps` or
   :doc:`../../management/shipments_deliveries/delivery_three_steps`.
```

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
-date they entered the stock. Here, a demand for some products triggers a removal rule that requests a
-transfer for the lot/serial number that has entered your stock the last.
+Similar to the :abbr:`FIFO (First In, First Out)` method, the :guilabel:`Last In, First Out (LIFO)`
+removal strategy moves products based on the date they entered a warehouse's stock.  Instead of
```
Remove extra space
```suggestion
removal strategy moves products based on the date they entered a warehouse's stock. Instead of
```

samueljlieber commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
-Next, you need to define your removal strategy, on *Product Categories* via
-:menuselection:`Inventory --> Configuration --> Product Categories`.
+To view all products with lots/serial numbers assigned to them, navigate to
+:menuselection:`Inventory app --> Products  --> Lots/Serial Numbers`. This reveals a page with
```
Remove extra space
```suggestion
:menuselection:`Inventory app --> Products --> Lots/Serial Numbers`. This reveals a page with
```

samueljlieber reviewed (approved):
Hi @larm-odoo! Nice job with implementing the improvements, I caught just a few more corrections. Please implement these before tagging final review! Approving now, nice job 🙂

larm-odoo commented:
Thanks @samueljlieber! Moving it along now to @StraubCreative.

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
-and they don’t have the same expiration date. In that situation, you logically prefer sending those
-with the closest date first. Depending on the removal strategy you chose, Odoo generates a transfer
-with the products fitting your settings the best.
+Next, several orders for the same product are made, but in this example, the goods weren't received
```
Transition word "Next" implies there's more directions to receiving products in Odoo...don't think that's what's happening here?

What kind of orders?
We just received products didn't we? ...so why would new orders be created?

If this is a new layer in the hypothetical, then explicitly say so 😉

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
-with the closest date first. Depending on the removal strategy you chose, Odoo generates a transfer
-with the products fitting your settings the best.
+Next, several orders for the same product are made, but in this example, the goods weren't received
+on the same day, and they don't have the same expiration date. In that situation, logically, sending
+the products with the closest expiration date is preferred. Depending on the chosen removal
```
Assuming the products have an expiration date? We haven't really fleshed out the example that far to indicate this., e.g. "Imagine we are receiving cartons of eggs..."

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
+===============================
+
+Removal strategies determine which products are taken from the warehouse when orders are confirmed.
+This is why selecting the correct removal strategies to best suit the needs of the warehouse is
+important.
```
Unnecessary
```suggestion
```

larm-odoo replied:
Agreed- in a review it was suggested to add an intro to these so I added it. Happily removing =)

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
 
-When using a **First In, First Out** strategy, a demand for some products triggers a removal rule
-which requests a transfer for the lot/serial number that has entered your stock the first.
+When using a :guilabel:`First In, First Out (FIFO)` strategy, a demand for a product triggers a
```
grammar
"a demand" is like saying "an eCommerce" imo

```suggestion
When using a :guilabel:`First In, First Out (FIFO)` strategy, demand for a product triggers a
```

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
-When using a **First In, First Out** strategy, a demand for some products triggers a removal rule
-which requests a transfer for the lot/serial number that has entered your stock the first.
+When using a :guilabel:`First In, First Out (FIFO)` strategy, a demand for a product triggers a
+removal rule, which requests a transfer for the lot/serial number that entered the stock first (have
+been in stock for the longest time).
```
```suggestion
removal rule, which requests a transfer for the lot/serial number that entered the stock first (and therefore, has
been in stock for the longest time).
```

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
+removal rule, which requests a transfer for the lot/serial number that entered the stock first (have
+been in stock for the longest time).
+
+For example, imagine there are three lots of nails in the warehouse. Those three have the following
+lot numbers: 00001, 00002, 00003, each with five boxes of nails in it.
```
```suggestion
For example, imagine there are three lots of nails in the warehouse. Those three lots have the following
identification numbers: 00001, 00002, 00003, each with five boxes of nails in it.
```

OR

```suggestion
For example, imagine there are three lots of nails in the warehouse, and have the corresponding:
lot numbers: 00001, 00002, 00003. Each lot has five boxes of nails in it.
```

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
-you 6 boxes on the 11th of June. With the *FIFO* strategy selected, a transfer is requested for the
-five boxes of 00001 and one of the boxes in 00002 because 00001 has entered your stock before the
-others. The box from 00002 is taken because it has the oldest enter date after 00001.
+Lot 00001 entered the stock on May 23, lot 00002 on May 25, and lot 00003 on June 1. A customer
```
these are borderline input values, consider surrounding all of these instances in backticks
```suggestion
Lot `00001` entered the stock on May 23, lot `00002` on May 25, and lot `00003` on June 1. A customer
```

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
 
-So, for every order of a product with the *FIFO* strategy selected, Odoo requests a transfer for the
-good that has been in your stock for the longest period.
+Using the :abbr:`FIFO (First In, First Out)` removal strategy, a transfer is requested for the five
+boxes from lot 00001 and one of the boxes in lot 00002, since lot 00001 entered the stock first. The
+box from lot 00002 is taken next because it has the oldest receipt date after lot 00001.
```
specificity/language
```suggestion
Using the :abbr:`FIFO (First In, First Out)` removal strategy, a transfer request will pick the five
boxes from lot 00001 first, and then from the boxes in lot 00002, since lot 00001 entered the stock first. The
box from lot 00002 is taken next because it has the oldest receipt date after lot 00001.
```

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
 
-So, basically, for every order of a product with the *LIFO* strategy used, a transfer for the last
-one to have entered the stock is requested.
+.. Warning::
```
lowercase admonition
```suggestion
.. warning::
```

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
-.. note::
-   This strategy is banned in many countries and can lead to only have old or obsolete products
-   in your stock.
+For example, imagine there are three lots of screws in the warehouse. Those three have the following
+numbers: 10001, 10002, and 10003, each with 10 boxes of screws in it.
```
double-check wording/grammar

StraubCreative reviewed (changes requested):
Hi @larm-odoo 
Thorough as always 😉 
I have some comments below mostly around wording and grammar along with some smaller nuances for your consideration. I didn't highlight every instance— can you take a look please?
Thanks and tag me again when we're ready for another look.

larm-odoo commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,362 @@
+either the :guilabel:`Product Category` or the :guilabel:`Location`.
+
+To change the :guilabel:`Removal Strategy`, go to :menuselection:`Inventory app --> Configuration
+--> Locations` or :guilabel:`Product Categories`. Click on a :guilabel:`Location` or
```
I decided on (and hope you agree) that the 'locations' and 'product categories' in the previous sentence should both be menu selections since they are part of the navigation drop downs (truly a menu selection). Clicking the location or category in this sentence, they are items you click in the dashboard view, as oppose to a navigation direction- so since they are different ways of selecting, I think everything in the previous sentence is a menu selection, and these are GUI's. If you disagree, no problem- I can change again- but that is the logic I am following (so my first gui for 'product categories' was incorrect).

larm-odoo commented:
Hi @StraubCreative - I fixed this up according to your review. I did re-write the top section a touch so it made more sense. I saw where you wanted it to be more clear as to what was being said, so I removed one photo that didn't seem to work, and kept the example of "generic product" so that it was more clear it was a general description. I think it works, but am of course open to more changes or suggestions if it's not what you had in mind. Also, I am second guessing myself, but I added backticks to all my lot numbers. I don't know if it was overkill, but where you suggested to add it, it looked like it made sense to apply it to all of them, since they are called out in all the images/paperwork. I am honestly not sure about that- at the time I was 100% positive, but now I am closer to 90%, so I would like your input on that! So other than those 2 things, I think it's all polished up!

StraubCreative commented on content/applications/inventory_and_mrp/inventory/routes/strategies/removal.rst:
```diff
@@ -1,222 +1,370 @@
-the transfer. The way Odoo assigns these products depends on the *Removal Strategy* defined in
-the *Product Category* or on the *Location*.
+=====================================
+Removal Strategies (FIFO, LIFO, FEFO)
```
sentence case 😉 
```suggestion
Removal strategies (FIFO, LIFO, FEFO)
```

StraubCreative reviewed (approved):
look good @larm-odoo 👍 
I'm going to push up the h1 change in a moment...
@samueljlieber can you fix the capital letters in the media filenames, please?

@robodoo delegate=samueljlieber

StraubCreative commented:
squashed commits + fixed commit message on [cb723b4](https://github.com/odoo/documentation/pull/3783/commits/cb723b450a7fe7e942989f68aad7f126620339c1)

quick CR on [9a86f36](https://github.com/odoo/documentation/pull/3783/commits/9a86f364ada16a971dbf346cc0dba759027e00c7)

this is ready for you @samueljlieber 👍

samueljlieber commented:
Fixed _Co-authored-by_ tags in commit message in https://github.com/odoo/documentation/commit/79c9920b30242a19f5701df25eea24f19abb9e5d

Ready for merge– this PR is targeting 14.0 only and will not be FWP.

@fw-bot ignore

@robodoo r+
